tests/admin_views/test_related_object_lookups.py

```diff
@@ -76,3 +75,51 @@ def test_related_object_lookup_link_enabled(self):
             with self.subTest(link_id):
                 link = self.selenium.find_element(By.XPATH, f'//*[@id="{link_id}"]')
                 self.assertIsNone(link.get_attribute("aria-disabled"))
+        
+    def test_filter_horizontal_without_duplication(self):
+        """Test that the filter_horizontal widget doesn't duplicate entries in the Chosen column after instance is added from another field using the plus JS action."""
+        # Use Transition and TransitionState models from tests.admin_views.models with the TransitionAdmin from tests.admin_views.admin
+        from selenium.webdriver.common.by import By
+
+        def _get_HTML_inside_element_by_id(id_):
+            return self.selenium.find_element(By.ID, id_).get_attribute("innerHTML")
+
+        transition_add_url = reverse("admin:admin_views_transition_add")
+        self.selenium.get(self.live_server_url + transition_add_url)
+        state_name = "test state"
+
+        # Add a TransitionState from the plus button on the Target field
+        self.selenium.find_element(By.ID, "add_id_target").click()
+        
+        # Switch to the popup window
+        self.wait_for_and_switch_to_popup()
+
+        # Find the label field and add text to it
+        self.selenium.find_element(By.ID, "id_label").send_keys(state_name)
+
+        # Now find the save button and click it
+        self.selenium.find_element(By.NAME, "_save").click()
+
+        # Return to the main window
+        self.wait_until(lambda d: len(d.window_handles) == 1, 1)
+        self.selenium.switch_to.window(self.selenium.window_handles[0])
+
+        # Now check that the state shows in the Target box
+        self.assertHTMLEqual(
+            _get_HTML_inside_element_by_id("id_target"),
+            f"""
+            <option value="" selected="">---------</option>
+            <option value="1" selected>{state_name}</option>
+            """
+        )
+
+        # Check that the state is in the Available Source box
+        self.assertHTMLEqual(
```

Comment:
This section is failing currently. I would assume that we would expect this to be filled with the new value, and that we only prevent it from showing in the "Chosen" column. Not sure where to resolve this, maybe in `SelectFilter2.js`. Open to direction on this, or if it's a pre-existing bug then we can just move forward.

Reply:
<img width="620" alt="image" src="https://github.com/django/django/assets/36718613/31918f61-ce1f-4e99-9705-230bfb8cf9ef">